zookeeper-api/src/test/java/org/apache/helix/zookeeper/impl/client/TestRawZkClient.java

```diff
@@ -868,59 +864,38 @@ public void testAsyncWriteOperations() {
-    // Remove "final" modifier
-    Field modifiersField = Field.class.getDeclaredField("modifiers");
-    boolean isModifierAccessible = modifiersField.isAccessible();
-    modifiersField.setAccessible(true);
-
-    Field packetLenField = ClientCnxn.class.getDeclaredField("packetLen");
-    Field childrenLimitField =
-        org.apache.helix.zookeeper.zkclient.ZkClient.class.getDeclaredField("NUM_CHILDREN_LIMIT");
-    modifiersField.setInt(packetLenField, packetLenField.getModifiers() & ~Modifier.FINAL);
-    modifiersField.setInt(childrenLimitField, childrenLimitField.getModifiers() & ~Modifier.FINAL);
-
-    boolean isPacketLenAccessible = packetLenField.isAccessible();
-    packetLenField.setAccessible(true);
-    int originPacketLen = packetLenField.getInt(null);
-    // Keep 150 bytes for successfully creating each child node.
-    packetLenField.set(null, 150);
-
-    boolean isChildrenLimitAccessible = childrenLimitField.isAccessible();
-    childrenLimitField.setAccessible(true);
-    int originChildrenLimit = childrenLimitField.getInt(null);
-    childrenLimitField.set(null, 2);
-
-    String path = "/" + TestHelper.getTestMethodName();
-    // Create 5 children to make packet length of children exceed 150 bytes
+    // Create 110K children to make packet length of children exceed 4 MB
     // and cause connection loss for getChildren() operation
-    for (int i = 0; i < 5; i++) {
-      _zkClient.createPersistent(path + "/" + UUID.randomUUID().toString(), true);
+    String path = "/" + TestHelper.getTestMethodName();
+    int toCreateChildrenCount = 110;
+
+    _zkClient.createPersistent(path);
+
+    for (int i = 0; i < toCreateChildrenCount; i++) {
+      List<Op> ops = new ArrayList<>(1000);
+      for (int j = 0; j < 1000; j++) {
+        String child = path + "/" + UUID.randomUUID().toString();
+        ops.add(Op.create(child, new byte[0], ZooDefs.Ids.OPEN_ACL_UNSAFE, CreateMode.EPHEMERAL));
+      }
+      // Reduce total creation time by batch creating znodes
+      _zkClient.multi(ops);
     }
 
     try {
       _zkClient.getChildren(path);
-      Assert.fail("Should not successfully get children.");
+      Assert.fail("Should not successfully get children because of connection loss.");
     } catch (ZkException expected) {
       Assert.assertEquals(expected.getMessage(),
           "org.apache.zookeeper.KeeperException$MarshallingErrorException: "
               + "KeeperErrorCode = MarshallingError");
     } finally {
-      packetLenField.set(null, originPacketLen);
-      packetLenField.setAccessible(isPacketLenAccessible);
-
-      childrenLimitField.set(null, originChildrenLimit);
-      childrenLimitField.setAccessible(isChildrenLimitAccessible);
-
-      modifiersField.setAccessible(isModifierAccessible);
+      _zkClient.close();
+      _zkClient = new ZkClient(ZkTestBase.ZK_ADDR);
 
       Assert.assertTrue(TestHelper.verify(() -> {
-        _zkClient.deleteRecursively(path);
+        _zkClient.delete(path);
```

Comment:
can we delete parent path here without child node deleted?

Reply:
No, we cannot because there would be node not empty exception.

Reply:
My concern is this:

if using deleteRecursively(), it won't work because list all the children znode name would cause "marshalling" error as what happened in GCN, right?

if using delete as you do here, I think it would throw another exception as not empty. So this does not work here. Or am I missing something?

Reply:
delete() returns boolean. Should handle `ZkException` here as I've removed deleteRecursively().

Reply:
So in this case, it would always throw exception, right? 

Then why bother. Just leave the znode there anyway.

Reply:
As a good practice, a unit test should cleanup itself and avoid other tests.
It won't always close. If you carefully understand the code, you'll find the tips of using ephemeral nodes and closing the zkClient to delete the ephemeral nodes for cleanup.

Reply:
My question is that is this exception always thrown? I believe it would always be thrown. Can you verify?

Reply:
Can we also re-open this comment thread?

Reply:
What do you mean by `always thrown`?

Reply:
`String path = "/" + TestHelper.getTestMethodName()` is path.

Now at the time this line of code `_zkClient.delete(path);` , there are a lot of znode underlying this `path`. So the delete should always fail, right?  Or am I understanding this wrong?

Reply:
@kaisun2000 I've replied you: they are ephemeral znodes.
> It won't always throw exception. If you carefully understand the code, you'll find the tips of using ephemeral nodes and closing the zkClient to delete the ephemeral nodes for cleanup. The exception handling here is just in case zk server doesn't remove ephemeral nodes yet.

Reply:
I see. Did not notice this ephemeral node. So closing the _zkClient delete the nodes. The second delete is just to delete the parent path. This looks good.